cmd/ipfs/init.go

```diff
@@ -49,19 +48,17 @@ var initCmd = &cmds.Command{
 			res.SetError(err, cmds.ErrNormal)
 			return
 		}
+
 		if !bitsOptFound {
 			nBitsForKeypair = nBitsForKeypairDefault
 		}
 
-		rpipe, wpipe := io.Pipe()
-		go func() {
-			defer wpipe.Close()
-			if err := doInit(wpipe, req.Context().ConfigRoot, force, nBitsForKeypair); err != nil {
-				res.SetError(err, cmds.ErrNormal)
-				return
-			}
-		}()
-		res.SetOutput(rpipe)
+		var buf bytes.Buffer
+		if err := doInit(&buf, req.Context().ConfigRoot, force, nBitsForKeypair); err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+		res.SetOutput(&buf)
```

Comment:
In general I agree but I would urge you to make this consideration on a case by case basis.

The output of `ipfs init` is small, constant and only used once and IMHO the pipe makes the code convoluted without any gain over that.

BTW: The implementation of `io.Pipe` still uses a byte slice to buffer when it can't write through immediately. If you really wan't this you would want an `os.Pipe` instead, which uses the pipe syscall to get two FDs from the kernel.